engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -355,4 +361,66 @@ private static boolean isErrorLevelEnable()
     return false;
   }
 
+  /**
+   * Adds Logger Appender
+   * @param name Appender name
+   * @param className Appender class name
+   * @param args Appender args
+   */
+  public static void addAppender(String name, String className, Object args)
+  {
+    if (getAppendersNames().contains(name)) {
+      logger.warn("A logger appender with the name '{}' exists. Cannot add a new logger appender with the same name", name);
+    } else {
+      try {
+        Constructor<?> constructor = Class.forName(className).getConstructor(Object.class);
```

Comment:
Is there a need to add that constructor restriction? Isn't the data that is passed specific to the appender implementation. Take a look at com.datatorrent.api.StringCodec.Object2String